atomic_reactor/plugins/pre_flatpak_update_dockerfile.py

```diff
@@ -0,0 +1,257 @@
+            composes = resolve_comp_result['composes']
+        else:
+            # But in a worker, resolve_composes doesn't run, so we have
+            # to load the compose info ourselves
+            assert self.compose_ids
+
+            composes = self._load_composes()
+
+        for compose_info in composes:
+            if compose_info['source_type'] != SOURCE_TYPE_MODULE:
+                continue
+
+            modules = [ModuleSpec.from_str(s) for s in compose_info['source'].split()]
+            for module in modules:
+                if module.name == main_module.name and module.stream == main_module.stream:
+                    return self._build_compose_info(modules)
+
+        self.log.debug('Compose info: %s', composes)
+        raise RuntimeError("Can't find main module %s in compose result" % source_spec)
+
+    def _load_source(self):
+        flatpak_yaml = self.workflow.source.config.flatpak
+
+        compose_info = get_flatpak_compose_info(self.workflow)
+
+        module_spec = ModuleSpec.from_str(compose_info.source_spec)
+
+        return FlatpakSourceInfo(flatpak_yaml,
+                                 compose_info.modules,
+                                 compose_info.main_module,
+                                 module_spec.profile)
+
+    def run(self):
+        """
+        run the plugin
+        """
+
+        compose_info = self._load_compose_info()
+        set_flatpak_compose_info(self.workflow, compose_info)
+
+        source = self._load_source()
+        set_flatpak_source_info(self.workflow, source)
+
+        builder = FlatpakBuilder(source, None, None)
+
+        builder.precheck()
+
+        # Update the dockerfile
+
+        # We need to enable all the modules other than the platform pseudo-module
+        enable_modules_str = ' '.join(builder.get_enable_modules())
+
+        install_packages_str = ' '.join(builder.get_install_packages())
+
+        replacements = {
+            '@ENABLE_MODULES@': enable_modules_str,
+            '@INSTALL_PACKAGES@': install_packages_str,
+            '@RELEASE@': compose_info.main_module.version,
+        }
+
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        content = dockerfile.content
+
+        # Perform the substitutions; simple approach - should be efficient enough
+        for old, new in replacements.items():
+            content = content.replace(old, new)
+
+        dockerfile.content = content
+
+        # Create a file describing which packages from the base yum repositories are included
+
+        includepkgs = builder.get_includepkgs()
+        includepkgs_path = os.path.join(self.workflow.builder.df_dir, 'atomic-reactor-includepkgs')
+        with open(includepkgs_path, 'w') as f:
+            f.write('includepkgs = ' + ','.join(includepkgs) + '\n')
+
+        # Create the cleanup script
+
+        cleanupscript = os.path.join(self.workflow.builder.df_dir, "cleanup.sh")
```

Comment:
Can cleanup.sh be a constant instead of a repeated magic variable string?

Reply:
Sure, added: